notebooks/flusight_from_idata.qmd

```diff
@@ -0,0 +1,233 @@
+
+```{python}
+# examine the observed data (dimensions, coordinates, data variables)
+idata.observed_data
+```
+
+```{python}
+# | label: examine-observed-data
+import matplotlib.pyplot as plt
+
+
+# examine the observations
+fig, ax = plt.subplots(1, 1, figsize=(8,6))
+data = idata.observed_data["obs"].to_numpy()
+ax.plot(
+    list(range(len(data))),
+    data,
+    color="black",
+    linewidth=1.0,
+    markersize=3.0,
+    marker="o"
+)
+ax.set_ylabel("Hospital Admissions")
+ax.set_xlabel("Time (Days)")
+plt.show()
+```
+
+Observation 1: By default, dates are not associated with any group. As such, either a list of dates corresponding to fitting and forecast periods must be generated from the initial and final fitting dates and then integrated into a dataframe OR dates must be added to the `idata` object itself.
+
+Observation 2: The name of the jurisdiction that the inference data object is associated with is not contained in the `idata` object.
+
+For Observation 1, if the former path is pursued, we might expect `forecasttools` to have a function as such:
+
+```python
+def forecast_as_df_with_dates(
+    idata_wo_dates: az.InferenceData,
+    start_date: str = "2022-08-08",
+    end_date: str = "2023-12-08",
+    lookahead: int = 28,
+) -> pl.DataFrame:
+    """
+    Extract observed_data and posterior_predictive
+    data from idata object. Assumes variable name is
+    "obs" in either case. Use obs_dim_0, start_date,
+    and end_date to create dates column. Convert
+    forecast to polars dataframe. Then transform
+    dataframe to a dataset of posterior draws.
+    """
+    # code to perform
+    pass
+```
+
+And if the latter path is pursued:
+
+
+```python
+def add_dates_to_idata(
+    idata_wo_dates: az.InferenceData,
+    start_date: str ="2022-08-08",
+    end_date: str ="2023-12-08",
+    lookahead: int = 28,
+) -> az.InferenceData:
+    """
+    Use obs_dim_0, start_date, and end_date
+    to create dates list. Add dates list to
+    coordinates of observed_data and
+    posterior_predictive.
+    """
+    # code to perform
+    pass
+```
+
+Presently, `forecasttools` operates via the former pattern:
+
+```{python}
+forecast_df = forecasttools.forecast_as_df_with_dates(
+    idata,
+    start_date="2022-08-08",
+    end_date="2023-12-08",
+    location="TX"
+)
+forecast_df
+```
+
+Format to epiweeks:
+
+```{python}
+
+forecast_df = forecasttools.daily_to_epiweekly(
+    forecast_df,
+    value_col = "hosp",
+    id_cols = [".draw", "location"],
+    weekly_value_name = "weekly_hosp"
+)
+forecast_df
+```
+
+Add trajectories to quantiles:
+
+```{python}
+forecast_df = forecasttools.trajectories_to_quantiles(
+    forecast_df,
+    timepoint_cols = ["epiweek", "epiyear"],
+    id_cols = ["location"],
+    value_col = "weekly_hosp"
+)
+forecast_df
+```
+
+Recode locations:
+
+
+```{python}
+forecast_df = forecasttools.loc_abbr_to_flusight_code(
+    df=forecast_df, location_col="location")
+```
```

Comment:
Some file path is printed with this is called. That should be removed.

Reply:
Sounds good.

Reply:
This was fixed after using the below pattern (not exact code):

```python
new_df = example_df.with_columns(
    location=pl.col("location").replace(
        old=loc_table["short_name"], 
        new=loc_table["location_code"]
    )
)
```